homework/index.js

```diff
@@ -28,23 +28,56 @@
     });
     return elem;
   }
+  function renderRepoDetails(table, title, repoText, article, repoHref = '') {
+    const tr = createAndAppend('tr', table);
+    if (article === 'a') {
+      createAndAppend('th', tr, {
+        text: title,
+      });
 
-  function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+      createAndAppend(article, tr, {
+        text: repoText,
+        href: repoHref,
+      });
+    } else {
+      createAndAppend('th', tr, {
+        text: title,
+      });
+      createAndAppend(article, tr, { text: repoText });
+    }
+  }
+
+  function sortName(a, b) {
+    return a.name.localeCompare(b.name);
   }
 
   function main(url) {
     fetchJSON(url, (err, repos) => {
       const root = document.getElementById('root');
+      createAndAppend('header', root, {
+        text: 'HYF Repositories',
+        class: 'hyf-title',
+      });
+
       if (err) {
         createAndAppend('div', root, {
           text: err.message,
           class: 'alert-error',
         });
         return;
       }
-      const ul = createAndAppend('ul', root);
-      repos.forEach(repo => renderRepoDetails(repo, ul));
+      const ul = createAndAppend('ul', root, {
+        class: 'ul-root',
+      });
+      repos.sort(sortName).forEach(repo => {
```

Comment:
With the modified `repoDetails()` function, you can now replace lines 72-80 with these three simple lines:

```js
repos
  .sort((a, b) => a.name.localeCompare(b.name))
  .forEach(repo => renderRepoDetails(repo, ul));
```